src/action/ActionControl.kt

```diff
@@ -1,5 +1,12 @@
 package action
 
-enum class Action {
-    MOVE_NORTH
+enum class Action(val nextDirectionToTry: String) {
+    MOVE_NORTHEAST("MOVE_NORTH"),
+    MOVE_EAST("MOVE_NORTHEAST"),
+    MOVE_SOUTHEAST("MOVE_EAST"),
+    MOVE_SOUTH("MOVE_SOUTHEAST"),
+    MOVE_SOUTHWEST("MOVE_SOUTH"),
+    MOVE_WEST("MOVE_SOUTHWEST"),
+    MOVE_NORTHWEST("MOVE_WEST"),
+    MOVE_NORTH("MOVE_NORTHWEST");
```

Comment:
The Action class is a part of World API, but here you have used it to hardcode a very specific behavior for one specific player.
Moreover, since all these moves are different only in a direction, it's probably better to have a single MOVE action (also probaby a class rather than enum), that will store the move vector.